src/amo/components/SearchContextCard/index.js

```diff
@@ -117,15 +172,64 @@ export class SearchContextCardBase extends React.Component {
   }
 }
 
-export function mapStateToProps(state) {
+export function mapStateToProps(state: AppState) {
+  const { search } = state;
+  const { filters = {} } = search;
+
+  let currentCategory;
+  let categoryName = null;
+
+  if (
+    search &&
+    filters &&
+    filters.category &&
+    typeof filters.category === 'string'
+  ) {
+    currentCategory = filters.category;
+  }
+
+  if (currentCategory) {
+    const categoriesState = state.categories.categories;
+    const { clientApp } = state.api;
+
+    if (categoriesState && clientApp) {
+      const appTypes = categoriesState[clientApp];
+
+      if (appTypes) {
+        if (
+          filters &&
+          filters.addonType &&
+          typeof filters.addonType === 'string'
+        ) {
+          // eslint-disable-next-line prefer-destructuring
+          let addonType = filters.addonType;
+          if (addonType === ADDON_TYPE_THEMES_FILTER) {
+            addonType = ADDON_TYPE_STATIC_THEME;
+          }
+
+          const categories = appTypes[addonType];
+
+          if (categories) {
+            categoryName = categories[currentCategory].name;
```

Comment:
`categories[currentCategory]` could be `undefined` here, so I think you need `categoryName = categories[currentCategory] ? categories[currentCategory].name || null;`. Maybe there should also be a test for that, which would have caught this?

Reply:
hmm let me recheck

Reply:
This still doesn't seem right to me. I don't think the test you added is addressing this use case because your test loads no categories, so `categories` will be falsey above, but when you want to emulate in the test is `categories` being true, but `categories[currentCategory]` returning `undefined`, because the specific category requested doesn't exist. You can test this interactively by applying your patch and going to http://localhost:3000/en-US/firefox/search/?category=bookmarksx&sort=rating&type=extension